src/cpp/session/modules/SessionPath.cpp

```diff
@@ -116,39 +121,46 @@ Error initialize()
 
    }
 
-   // build the PATH
-   std::vector<std::string> parts =
-         core::algorithm::split(core::system::getenv("PATH"), ":");
-
    // add in components from paths.d etc.
    std::for_each(paths.begin(),
                  paths.end(),
-                 boost::bind(addToPathIfNecessary, _1, &parts));
+                 boost::bind(addToPathIfNecessary, _1, &paths));
```

Comment:
Wait, I'm confused about these lines. We're looping through paths, and then adding paths to paths? Isn't this completely unnecessary, now that we're not starting with parts anymore?

Reply:
You are correct -- let me take another look.

Reply:
Should be fixed up now; thanks for catching that!